src/core/buffers/period/period_buffer.ts

```diff
@@ -114,18 +114,22 @@ export default function PeriodBuffer({
   const adaptation$ = new ReplaySubject<Adaptation|null>(1);
   return adaptation$.pipe(
     switchMap((adaptation) => {
-      if (adaptation == null) {
+      if (adaptation === null) {
         log.info(`Buffer: Set no ${bufferType} Adaptation`, period);
-        const previousQSourceBuffer = sourceBuffersStore.get(bufferType);
+        const sourceBufferStatus = sourceBuffersStore.getStatus(bufferType);
         let cleanBuffer$ : Observable<unknown>;
 
-        if (previousQSourceBuffer != null) {
+        if (sourceBufferStatus.type === "set") {
           log.info(`Buffer: Clearing previous ${bufferType} SourceBuffer`);
-          cleanBuffer$ = previousQSourceBuffer
+          cleanBuffer$ = sourceBufferStatus.value
             .removeBuffer(period.start,
                           period.end == null ? Infinity :
                                                period.end);
+          return clock$.pipe(map(tick => EVENTS.needsMediaSourceReload(tick)));
         } else {
+          if (sourceBufferStatus.type === "unset") {
```

Comment:
It's a noobies question but how the information on `sourceBufferStatus` arrive at this level, to know the source buffer status ?

Reply:
Since a recent refactoring, you are supposed to use the `SourceBufferStore` not only to create a SourceBuffer, but also to announce that you won't be using one for a given type (ex: no video SourceBuffer).

Based on this information, the SourceBufferStore now knows if a SourceBuffer has been created or disabled. It has three states for each possible types of SourceBuffer:
  - `"set`": The SourceBuffer  has been created. It knows that because you had to call its `createSourceBuffer` method to do that.
  - `"disabled"`: The SourceBuffer has been explicitely disabled. Here it knows that because you had to call its `disableSourceBuffer` method.
  - `"unset"`: The SourceBuffer status is yet unknown. This is the case when none of those two APIs have yet been called for the type.

PS: The SourceBufferStore then has an API to emit an event when all "native" SourceBuffers (audio+video) have either been created or disabled (it is needed that at least one of those is created though).

Reply:
Thanks!